components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpoint.java

```diff
@@ -67,202 +71,188 @@
-                        for (int i = 0; i < headers.length; i++) {
-                            if (headers[i] != null) {
-                                respBuilder.header(headers[i].getKey(), headers[i].getValue());
-                            }
-                        }
-                    }
+            return respBuilder.entity(response.getBody()).build();
+        }
+    }
 
-                    return respBuilder.entity(response.getBody()).build();
-                }
-            } else {
-                OAuthTokenResponseBuilder oAuthRespBuilder = OAuthASResponse
-                        .tokenResponse(HttpServletResponse.SC_OK)
-                        .setAccessToken(oauth2AccessTokenResp.getAccessToken())
-                        .setRefreshToken(oauth2AccessTokenResp.getRefreshToken())
-                        .setExpiresIn(Long.toString(oauth2AccessTokenResp.getExpiresIn()))
-                        .setTokenType(BEARER);
-                oAuthRespBuilder.setScope(oauth2AccessTokenResp.getAuthorizedScopes());
-
-                // OpenID Connect ID token
-                if (oauth2AccessTokenResp.getIDToken() != null) {
-                    oAuthRespBuilder.setParam(OAuthConstants.ID_TOKEN,
-                            oauth2AccessTokenResp.getIDToken());
-                }
-                OAuthResponse response = oAuthRespBuilder.buildJSONMessage();
-                ResponseHeader[] headers = oauth2AccessTokenResp.getResponseHeaders();
-                ResponseBuilder respBuilder = Response
-                        .status(response.getResponseStatus())
-                        .header(OAuthConstants.HTTP_RESP_HEADER_CACHE_CONTROL,
-                                OAuthConstants.HTTP_RESP_HEADER_VAL_CACHE_CONTROL_NO_STORE)
-                        .header(OAuthConstants.HTTP_RESP_HEADER_PRAGMA,
-                                OAuthConstants.HTTP_RESP_HEADER_VAL_PRAGMA_NO_CACHE);
-
-                if (headers != null && headers.length > 0) {
-                    for (int i = 0; i < headers.length; i++) {
-                        if (headers[i] != null) {
-                            respBuilder.header(headers[i].getKey(), headers[i].getValue());
-                        }
-                    }
-                }
+    private String getConsumerKey(HttpServletRequestWrapper httpRequest) {
+        return httpRequest.getParameter(OAuth.OAUTH_CLIENT_ID);
+    }
+
+    private void validateAuthorizationHeader(HttpServletRequest request, MultivaluedMap<String, String> paramMap)
+            throws OAuthSystemException, TokenEndpointAccessDeniedException {
 
-                return respBuilder.entity(response.getBody()).build();
+        try {
+            // The client MUST NOT use more than one authentication method in each request
+            if (isClientCredentialsExistsAsParams(paramMap)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Used more than one authentication method ");
```

Comment:
can we log "Used more than one authentication method . Client_ID and Client_secret both used in the request parameters" ?

Reply:
changed the log. Client Id and Client Secret found in request body and Authorization header. Credentials should be sent in either request body or Authorization header, not bot